urdf/ur_macro.xacro

```diff
@@ -278,18 +281,33 @@
       </xacro:if>
       <dynamics damping="0" friction="0"/>
     </joint>
-    <joint name="${tf_prefix}wrist_3_joint" type="revolute">
-      <parent link="${tf_prefix}wrist_2_link" />
-      <child link="${tf_prefix}wrist_3_link" />
-      <origin xyz="${wrist_3_x} ${wrist_3_y} ${wrist_3_z}" rpy="${wrist_3_roll} ${wrist_3_pitch} ${wrist_3_yaw}" />
-      <axis xyz="0 0 1" />
-      <limit lower="${wrist_3_lower_limit}" upper="${wrist_3_upper_limit}"
-             effort="${wrist_3_effort_limit}" velocity="${wrist_3_velocity_limit}"/>
-      <xacro:if value="${safety_limits}">
-         <safety_controller soft_lower_limit="${wrist_3_lower_limit + safety_pos_margin}" soft_upper_limit="${wrist_3_upper_limit - safety_pos_margin}" k_position="${safety_k_position}" k_velocity="0.0"/>
-      </xacro:if>
-      <dynamics damping="0" friction="0"/>
-    </joint>
+    <xacro:if value="${sec_limits['wrist_3_joint']['has_position_limits']}">
+      <joint name="${tf_prefix}wrist_3_joint" type="revolute">
+        <parent link="${tf_prefix}wrist_2_link" />
+        <child link="${tf_prefix}wrist_3_link" />
+        <origin xyz="${wrist_3_x} ${wrist_3_y} ${wrist_3_z}" rpy="${wrist_3_roll} ${wrist_3_pitch} ${wrist_3_yaw}" />
+        <axis xyz="0 0 1" />
+        <limit lower="${wrist_3_lower_limit}" upper="${wrist_3_upper_limit}"
+              effort="${wrist_3_effort_limit}" velocity="${wrist_3_velocity_limit}"/>
+        <xacro:if value="${safety_limits}">
+          <safety_controller soft_lower_limit="${wrist_3_lower_limit + safety_pos_margin}" soft_upper_limit="${wrist_3_upper_limit - safety_pos_margin}" k_position="${safety_k_position}" k_velocity="0.0"/>
+        </xacro:if>
+        <dynamics damping="0" friction="0"/>
+      </joint>
+    </xacro:if>
+    <xacro:unless value="${sec_limits['wrist_3_joint']['has_position_limits']}">
+      <joint name="${tf_prefix}wrist_3_joint" type="continuous">
+        <parent link="${tf_prefix}wrist_2_link" />
+        <child link="${tf_prefix}wrist_3_link" />
+        <origin xyz="${wrist_3_x} ${wrist_3_y} ${wrist_3_z}" rpy="${wrist_3_roll} ${wrist_3_pitch} ${wrist_3_yaw}" />
+        <axis xyz="0 0 1" />
+        <limit effort="${wrist_3_effort_limit}" velocity="${wrist_3_velocity_limit}"/>
+        <xacro:if value="${safety_limits}">
+          <safety_controller k_position="${safety_k_position}" k_velocity="0.0"/>
+        </xacro:if>
+        <dynamics damping="0" friction="0"/>
+      </joint>
+    </xacro:unless>
```

Comment:
I think, this can be written a bit neater:

- We could specify the joint_type, e.g. in `ur_common` as `wrist_3_joint_type`
- We could put a  `<xacro:if value="${wrist_3_joint_type} != continuous">` around the limit tag only.

IMHO this would reduce code duplication while also improving readability.